pkg/provider/k8scloudproviderprovider.go

```diff
@@ -164,12 +172,16 @@ func getMostCommonClusterProviderFromDistribution(d clusterProviderDistribution)
 }
 
 func getClusterProviderFromAnnotations(annotations map[string]string) (ClusterProvider, bool) {
-	const (
-		annotationNameGKEInstanceID = "container.googleapis.com/instance_id"
-	)
-
 	annotationsGKE := map[string]struct{}{
-		annotationNameGKEInstanceID: {},
+		"container.googleapis.com/instance_id": {},
+	}
+
+	annotationsK3S := map[string]struct{}{
+		"k3s.io/hostname":         {},
```

Comment:
Are those documented somewhere by K3s, or have you discovered them?

Reply:
Discovered in an already running cluster. Node YAML shared by @mheap 🙇